### PR TITLE
Key resolution optimizations and other minor updates

### LINKED ISSUE #83: Introduce key format and JWK 'kid' properties and optimize PublicKey creation
At the moment 'KeyLocationResolver' loads a public key from the external resource on every request. This can be mostly optimized which will need to be done anyway (have a local volatile PublicKey property should do it for PEM or single JWK files). 

But if it is a JWK set (which may contain N entries) then the problem is that the incoming 'kid' property can vary in theory so we can't settle on the single public key so iterating over the set every time can be expensive. In reality though it would be better and typically expected to have the same 'kid' - which would be managed at the key rotation point ex in the OIDC server. Expecting that every client may have used a different private key to sign is not really realistic from the security POV and as such the floating selection of keys from the JWK set can and should be optionally restricted to the specific 'kid' - and this will let it optimize the PublicKey creation as well.

Furthermore, with a new proiperty for specifying the key store format (PEM file, JKS store (in the future), JWK, JWKSet) the key selection can be further optimized.

It is related to #82 (we would not be able to do the above if the control of the JWK selection would stay inside Jose4J) so I'll fix both issues with a single PR

### PR DESCRIPTION
**Pull request by sberyozkin**
Fixes #83, #82, #81

All the key types are dealt with in `KeyLocationResolver` now so it will be easier to control it further, best attempt at resolving and calculating the public key only once.

### CODE REVIEW

File: implementation/src/main/java/io/smallrye/jwt/auth/principal/KeyLocationResolver.java [hunk 1]
```diff
@@ -45,77 +47,113 @@
  */
 public class KeyLocationResolver implements VerificationKeyResolver {
     private static final Logger log = Logger.getLogger(KeyLocationResolver.class);
-    private String location;
     private String content;
+    private HttpsJwks httpsJwks;
```
Comment (Ladicek) [inline implementation/src/main/java/io/smallrye/jwt/auth/principal/KeyLocationResolver.java]: Would be good to add a comment that `content` and `httpsJwks` are mutually exclusive.

**Comment by sberyozkin** [comment]
@kenfinnigan @Ladicek 

Thanks. Yes, it is not quite ideal because the HTTPs one is actually a cache which refreshes periodically, but at the moment I've just wanted to get all the key related code in one place, and we can improve it if proves possible during the next iterations

File: implementation/src/main/java/io/smallrye/jwt/auth/principal/KeyLocationResolver.java (hunk 1 above)
Comment (sberyozkin) [reply implementation/src/main/java/io/smallrye/jwt/auth/principal/KeyLocationResolver.java]: OK

**Comment by sberyozkin** [comment]
@kenfinnigan or @Ladicek can you please approve again and help me stop keeping updating the comments :-)

**Comment by sberyozkin** [comment]
@kenfinnigan and @Ladicek, going ahead with the merge as you did approve and I've updated the comments, so all should be good

**Comment by Ladicek** [comment]
Sure :-)

**Pull request merged**